### PR TITLE
Allow control of some serve configuration via env vars

### PR DESCRIPTION
**Pull request by timkpaine**
## Why are these changes needed?
When a serve app is launched, serve will startup automatically. In certain places like k8s, it can be difficult to preconfigure serve (e.g. in the [ray-cluster helm chart](https://github.com/ray-project/kuberay/blob/master/helm-chart/ray-cluster/values.yaml) there is no ability to set the default serve arguments). 

This means you need to either be explicit when you start serve, or if it starts up automatically you may need to shut it down, then restart it, which is inconvenient.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/serve/_private/constants.py [hunk 1]
```diff
@@ -16,19 +16,21 @@
...
 #: HTTP Host
-DEFAULT_HTTP_HOST = "127.0.0.1"
+DEFAULT_HTTP_HOST = os.environ.get("RAY_SERVE_DEFAULT_HTTP_HOST", "127.0.0.1")
 
 #: HTTP Port
-DEFAULT_HTTP_PORT = 8000
+DEFAULT_HTTP_PORT = int(os.environ.get("RAY_SERVE_DEFAULT_HTTP_PORT", 8000))
 
 #: Uvicorn timeout_keep_alive Config
 DEFAULT_UVICORN_KEEP_ALIVE_TIMEOUT_S = 5
```
Comment (timkpaine) [inline python/ray/serve/_private/constants.py:30]: there are more here and below, happy to expose all of them if it makes sense to do so

**Comment by GeneDer** [comment]
Lgtm!

File: python/ray/serve/_private/constants.py [hunk 2]
```diff
@@ -16,19 +16,21 @@
 SERVE_NAMESPACE = "serve"
 
 #: HTTP Address
-DEFAULT_HTTP_ADDRESS = "http://127.0.0.1:8000"
+DEFAULT_HTTP_ADDRESS = os.environ.get(
+    "RAY_SERVE_DEFAULT_HTTP_ADDRESS", "http://127.0.0.1:8000"
+)
```
Comment (edoakes) [inline python/ray/serve/_private/constants.py:21]: can you check what this one is used for and why it's different from the two below?

File: python/ray/serve/_private/constants.py (hunk 1 above)
Comment (edoakes) [reply python/ray/serve/_private/constants.py:30]: we can leave the surface area minimal for now

**Review by edoakes — APPROVED** [review]

File: python/ray/serve/_private/constants.py (hunk 2 above)
Comment (edoakes) [reply python/ray/serve/_private/constants.py:21]: @GeneDer can you follow up on this one? going to merge the PR

Comment (GeneDer) [reply python/ray/serve/_private/constants.py:21]: Oh interesting this one is actually not used anywhere. Let me follow up to just remove it

**Pull request merged**